src/coinbase/wallet.ts

```diff
@@ -211,9 +211,43 @@ export class Wallet {
       };
     }
     const response = await Coinbase.apiClients.address!.createAddress(this.model.id!, payload);
+    if (!this.addresses.length || !Coinbase.useServerSigner) {
+      this.reload();
+    }
+    const address = new WalletAddress(response.data, key);
+    this.addresses.push(address);
 
-    this.cacheAddress(response!.data, key);
-    return new WalletAddress(response!.data, key);
+    return address;
+  }
+
+  /**
+   * Lists the Wallets belonging to the User.
+   *
+   * @param pageSize - The number of Wallets to return per page. Defaults to 10
+   * @param nextPageToken - The token for the next page of Wallets
+   * @returns An object containing the Wallets and the token for the next page
+   */
+  public static async listWallets(
+    pageSize: number = 10,
+    nextPageToken?: string,
+  ): Promise<{ wallets: Wallet[]; nextPageToken: string }> {
+    const walletList = await Coinbase.apiClients.wallet!.listWallets(
+      pageSize,
+      nextPageToken ? nextPageToken : undefined,
+    );
+    const walletsModels: WalletModel[] = [];
+
+    for (const wallet of walletList.data.data) {
+      walletsModels.push(wallet);
+    }
+
+    const wallets = await Promise.all(
+      walletsModels.map(async wallet => {
+        return await Wallet.init(wallet, "");
+      }),
+    );
+
+    return { wallets: wallets, nextPageToken: walletList.data.next_page };
```

Comment:
Is there a reason we're doing this differently than `listTrades`?

I think this is fine until we figure out the best enumerator-like paradigm for TS, but ideally we're consistent across all of our paginated methods

Reply:
Yes, listTrades pulls all available data for a given address. On the other hand, listWallets only returns 10 records. I think Yuga implemented this function initially, we can implement the same approach for listWallets, and I can follow up with a PR for this.

Reply:
Cool, yeah the original implementation was a stop-gap that we removed in the ruby version. Can update this separately for consistency

Reply:
of course! I will.